save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Token.kt

```diff
@@ -0,0 +1,158 @@
+        div {
+            className = ClassName("col-2 text-right")
+            a {
+                href = "https://github.com/saveourtool/save-cloud/tree/master/save-api"
+                buttonBuilder(
+                    "API Readme",
+                    style = "outline-secondary rounded-pill btn-sm",
+                    isOutline = false
+                ) {
+                    }
+            }
+        }
+
+        div {
+            className = ClassName("col-2 text-left")
+            a {
+                href = "https://github.com/saveourtool/save-cloud/tree/master/save-api-cli"
+                buttonBuilder(
+                    "Usage example",
+                    style = "outline-secondary rounded-pill btn-sm",
+                    isOutline = false
+                ) {
+
+                    }
+            }
+        }
+    }
+
+    div {
+        className = ClassName("row justify-content-center mt-4")
+        div {
+            className = ClassName("col-5")
+            p {
+                style = jso {
+                    textAlign = TextAlign.justify
+                }
+                +TOKEN_TEXT
+            }
+        }
+    }
+
+    div {
+        className = ClassName("row justify-content-center mt-4")
+        div {
+            className = ClassName("col-6 text-center")
+            button {
+                type = ButtonType.button
+                className = ClassName("btn btn-primary")
+                +"Generate new token"
+                onClick = {
+                    setToken(generateToken())
+                    postToken(props.userInfo!!, token)
+                }
+            }
+        }
+    }
+
+    token?.let {
+        div {
+            className = ClassName("row justify-content-center mt-4")
+            div {
+                className = ClassName("col-6")
+                form {
+                    className = ClassName("form-group text-center")
+                    input {
+                        value = token
+                        required = true
+                        className = ClassName("form-control text-center")
+                    }
+                }
+                div {
+                    className = ClassName("invalid-feedback d-block text-center")
+                    +"This is your unique token. It will be shown to you only once! Please save it."
+                }
+            }
+        }
+    }
+}
+
+@Suppress("MAGIC_NUMBER")
+private fun generateToken(): String {
+    var token = "ghp_"
+    val charPool = ('a'..'z') + ('A'..'Z') + ('0'..'9')
+    while (token.length < 40) {
+        token += charPool.random()
+    }
+    return token
+}
+
+private fun postToken(userInfo: UserInfo, token: String?) {
+    useDeferredRequest {
+        post(
+            "$apiUrl/users/${userInfo.name}/save/token",
+            jsonHeaders,
+            token,
+            loadingHandler = ::loadingHandler,
+        )
+    }
+}
```

Comment:
You should not do so:
`useDeferredRequest` returns a callback to send a request, while `postToken` returns nothing. Moreover, it breaks the rule of hooks, as you use a hook not in highest level of `FC` or another hook.

Reply:
Yeah, my bad, cope-paste issues. 
It was a scope.launch there previously, so I made this mistake